scripts/tests/test_zkai_native_attention_mlp_variant_invariant_reprove_preflight_gate.py

```
@@ -0,0 +1,150 @@
+        if not hasattr(gate.os, "symlink"):
+            self.skipTest("symlink support required")
+        with tempfile.TemporaryDirectory() as temp_dir:
+            target = gate.pathlib.Path(temp_dir) / "target.json"
+            link = gate.JSON_OUT
+            backup = gate.JSON_OUT.read_bytes() if gate.JSON_OUT.exists() else None
+            target.write_text("{}", encoding="utf-8")
+            if link.exists():
+                link.unlink()
+            link.symlink_to(target)
+            try:
+                with self.assertRaisesRegex(gate.VariantInvariantReprovePreflightError, "symlink"):
```

Ruff (0.15.12)

[warning] Use `pytest.raises` instead of unittest-style `assertRaisesRegex`

Replace `assertRaisesRegex` with `pytest.raises`

(PT027)